bcda/cclf/cclf.go

```diff
@@ -509,87 +518,96 @@ func validate(fileMetadata *cclfFileMetadata, cclfFileValidator map[string]cclfF
+			log.Infof("Reading file %s from archive %s", fileMetadata.name, fileMetadata.filePath)
 		}
-		defer rc.Close()
-		sc := bufio.NewScanner(rc)
-		for sc.Scan() {
-			b := sc.Bytes()
-			bytelength := len(bytes.TrimSpace(b))
-			if bytelength > 0 && bytelength <= validator.maxRecordLength {
-				count++
-
-				// currently only errors if there are more records than we expect.
-				if count > validator.totalRecordCount {
-					fmt.Printf("Maximum record count reached for file %s, Expected record count: %d, Actual record count: %d.\n", key, validator.totalRecordCount, count)
-					err := fmt.Errorf("maximum record count reached for file %s (expected: %d, actual: %d)", key, validator.totalRecordCount, count)
-					log.Error(err)
-					return err
-				}
-			} else {
-				fmt.Printf("Incorrect record length for file %s, Expected record length: %d, Actual record length: %d.\n", key, validator.maxRecordLength, bytelength)
-				err := fmt.Errorf("incorrect record length for file %s (expected: %d, actual: %d)", key, validator.maxRecordLength, bytelength)
+	}
+
+	rc, err := rawFile.Open()
+	if err != nil {
+		fmt.Printf("Could not read file %s in archive %s.\n", fileMetadata.name, fileMetadata.filePath)
+		err = errors.Wrapf(err, "could not read file %s in archive %s", fileMetadata.name, fileMetadata.filePath)
+		log.Error(err)
+		return err
+	}
+	defer rc.Close()
+	sc := bufio.NewScanner(rc)
+	for sc.Scan() {
+		b := sc.Bytes()
+		bytelength := len(bytes.TrimSpace(b))
+		if bytelength > 0 && bytelength <= validator.maxRecordLength {
+			count++
+
+			// currently only errors if there are more records than we expect.
+			if count > validator.totalRecordCount {
+				fmt.Printf("Maximum record count reached for file %s, Expected record count: %d, Actual record count: %d.\n", key, validator.totalRecordCount, count)
+				err := fmt.Errorf("maximum record count reached for file %s (expected: %d, actual: %d)", key, validator.totalRecordCount, count)
 				log.Error(err)
 				return err
 			}
+		} else {
+			fmt.Printf("Incorrect record length for file %s, Expected record length: %d, Actual record length: %d.\n", key, validator.maxRecordLength, bytelength)
+			err := fmt.Errorf("incorrect record length for file %s (expected: %d, actual: %d)", key, validator.maxRecordLength, bytelength)
+			log.Error(err)
+			return err
 		}
 	}
 	fmt.Printf("Successfully validated CCLF%d file %s.\n", fileMetadata.cclfNum, fileMetadata)
 	log.Infof("Successfully validated CCLF%d file %s.", fileMetadata.cclfNum, fileMetadata)
 	return nil
 }
 
-func validateFileName(fileName string) error {
-	filenameRegexp := regexp.MustCompile(`(T|P).*\.((?:A|T)\d{4})\.ACO.*\.ZC(0|8)Y(\d{2})\.(D\d{6}\.T\d{6})\d`)
-	valid := filenameRegexp.MatchString(fileName)
-	if !valid {
-		fmt.Printf("Invalid filename for file: %s.\n", fileName)
-		err := fmt.Errorf("invalid filename for file: %s", fileName)
+func validateCCLFFolderName(folderName string) error {
+	// add comment for new folder format string
```

Comment:
I don't understand this comment.

Reply:
wanted to add something like this: 
// CCLF foldername convention for SSP with BCD identifier: P.BCD.A****.ZCY**.Dyymmdd.Thhmmsst
You'll see this in the next commit